src/org/zaproxy/zap/extension/proxies/OptionsProxiesPanel.java

```diff
@@ -0,0 +1,225 @@
+ * limitations under the License. 
+ */
+package org.zaproxy.zap.extension.proxies;
+
+import java.awt.GridBagLayout;
+import java.util.List;
+
+import javax.swing.JCheckBox;
+import javax.swing.JOptionPane;
+import javax.swing.SortOrder;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.extension.option.SecurityProtocolsPanel;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.view.AbstractParamPanel;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.utils.FontUtils;
+import org.zaproxy.zap.view.AbstractMultipleOptionsTablePanel;
+import org.zaproxy.zap.view.LayoutHelper;
+
+public class OptionsProxiesPanel extends AbstractParamPanel {
+
+    private static final long serialVersionUID = 1L;
+
+    private ProxiesMultipleOptionsPanel proxiesOptionsPanel;
+
+    private OptionsLocalProxyPanel mainProxyPanel;
+    private SecurityProtocolsPanel securityProtocolsPanel;
+    private OptionsProxiesTableModel proxiesModel;
+    private String currentAddress;
+    private int currentPort;
+
+    private ExtensionProxies extension;
+
+    public OptionsProxiesPanel(ExtensionProxies extension) {
+        super();
+        this.extension = extension;
+        initialize();
+    }
+
+    private void initialize() {
+        this.setName(Constant.messages.getString("proxies.options.title"));
+        this.setLayout(new GridBagLayout());
+        this.add(getMainProxyPanel(), LayoutHelper.getGBC(0, 0, 1, 0.0));
+        this.add(getSecurityProtocolsPanel(), LayoutHelper.getGBC(0, 1, 1, 0.0));
+        this.add(getProxiesOptionsPanel(), LayoutHelper.getGBC(0, 2, 1, 1.0, 1.0));
+    }
+
+    @Override
+    public void initParam(Object obj) {
+        OptionsParam optionsParam = (OptionsParam) obj;
+        ProxiesParam param = optionsParam.getParamSet(ProxiesParam.class);
+        getMainProxyPanel().setProxy(param.getMainProxy());
+        getSecurityProtocolsPanel().setSecurityProtocolsEnabled(param.getSecurityProtocolsEnabled());
+        getProxiesModel().setProxies(param.getProxies());
+        getProxiesOptionsPanel().setRemoveWithoutConfirmation(!param.isConfirmRemoveProxy());
+        this.currentAddress = param.getMainProxy().getAddress();
+        this.currentPort = param.getMainProxy().getPort();
+    }
+
+    @Override
+    public void validateParam(Object obj) throws Exception {
+        ProxiesParamProxy mainProxy = getMainProxyPanel().getProxy();
+        String newAddress = mainProxy.getAddress();
+        int newPort = mainProxy.getPort();
+
+        if (!this.currentAddress.equals(newAddress) || this.currentPort != newPort) {
+            // Only check if they've changed, otherwise we'll still be listening on them
+            if (!extension.canListenOn(newAddress, newPort) || extension.getAdditionalProxy(newAddress, newPort) != null) {
+                throw new Exception(Constant.messages.getString("options.proxy.dialog.proxy.warning.fail.message"));

```

Comment:
It should included the address/port, otherwise it's hard to know which one has problems.